File: SEGUNDO/PDOO/PRACTICA4/ruby4/Ruby/BetaPowerEfficientSpaceStation.rb
#encoding:utf-8

require 'singleton'

require_relative 'PowerEfficientSpaceStation.rb'
require_relative 'Dice.rb'

module Deepspace
    class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
        @@EXTRAEFFICIENCY = 1.20

        def initialize (station)
            super
            @dice = Dice.new
        end

        def fire
            if (dice.extraEfficiency)
                return (super * @@EXTRAEFFICIENCY)
            else
                return super
            end
        end
    end
end

File: SEGUNDO/PDOO/PRACTICAS/PRACTICA2/java/deepSpace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author lauralazaro
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
    private final float EXTRAEFFICIENCY = 1.2f;
    private Dice dice;
    
    BetaPowerEfficientSpaceStation(SpaceStation station) {
        super(station);
        dice=new Dice();
    }
    
    @Override public float fire() {
        
        if (dice.extraEfficiency())
            return (super.fire()*EXTRAEFFICIENCY);
        else
            return super.fire();
    }
}
